### DISCUSSION TITLE
Handling null values in Milvus

### DISCUSSION THREAD

**Discussion by Mahanchello** [milvus-io/milvus#22447] [Ideas & Feature requests]
Hey, just wanted to suggest the feature on **adding handling for `null`, `None`, `NaN` values in Milvus.** 
There are workarounds like storing an empty string, `-1` for ints, but could be really cool if Milvus would handle such values as well without erroring out on unknown DataType when inserting the data.  
Thank you!

**Comment by xiaofan-luan** [comment]
Ahh you r right. Null support is definitely important on some cases.
I'm planning to support it on 2.4 but if anyone want to take it we can make it happen faster!

**Comment by jiaoew1991** [comment]
@smellthemoon and I are working on that, if you have some suggestion we can discuss here. 😊

**Comment by hakan458** [comment]
I just ran into this issue as well. I am inserting DataFrames in batches and I am hitting an exception when any row in a column has a null value. For some reason that causes the data type returned in the mapping to be UNKNOWN. 
Right now I am going to handle this on my end using [fillna()](https://spark.apache.org/docs/3.1.1/api/python/reference/api/pyspark.sql.DataFrame.fillna.html#pyspark.sql.DataFrame.fillna) on my DataFrames but it would be great if Milvus could handle this! 

@jiaoew1991 @smellthemoon have you made any progress on this? I could possibly help as well

**Comment by xiaofan-luan** [comment]
Good catch! 

We are working on this feature for a while, should be able to release it on 2.4?

@smellthemoon @jiaoew1991

**Reply by smellthemoon** [reply]
Have done some work before. The plan is to implement null on stroage.

**Reply by 404-P4rziv4L** [reply]
Any updates on this feature?

**Reply by junjie-landing** [reply]
any update?

**Reply by xiaofan-luan** [reply]
there are already some pr for null support  
https://github.com/milvus-io/milvus/pull/32183
https://github.com/milvus-io/milvus/pull/33602
The feature will be released with 2.5

**Reply by arrrrny** [reply]
same here. definitely need this

**Reply by smellthemoon** [reply]
> same here. definitely need this

releted issue：#31728
